dvc_webdav/bearer_auth_client.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import shlex
+import subprocess
+import sys
+import threading
+from typing import Optional, Union
+
+import httpx
+
+logger = logging.getLogger("dvc")
+
+
+def _log_with_thread(level: int, msg: str, *args) -> None:
+    """
+    Universal helper to inject thread identity into logs.
+    Output format: [Thread-Name] Message...
+    """
+    if logger.isEnabledFor(level):
+        thread_name = threading.current_thread().name
+        log_fmt = f"[{thread_name}] " + msg
+        logger.log(level, log_fmt, *args)
+
+
+def execute_command(command: Union[list[str], str], timeout: int = 10) -> str:
+    """Executes a command to retrieve the token."""
+    if isinstance(command, str):
+        command = shlex.split(command)
+
+    try:
+        result = subprocess.run(  # noqa: S603
+            command,
+            shell=False,
+            capture_output=True,
+            text=True,
+            check=True,
+            timeout=timeout,
+            encoding="utf-8",
+        )
+    except (
+        FileNotFoundError,
+        subprocess.TimeoutExpired,
+        subprocess.CalledProcessError,
+        ValueError,
+        OSError,
+    ) as e:
+        error_header = "\n" + "=" * 60
+        error_msg = (
+            f"{error_header}\n[CRITICAL] Bearer Token Retrieval Failed.\n"
+            "DVC may misinterpret this as 'File Not Found' and skip files.\n"
+            f"Command: {command}\n"
+            f"Error: {e}"
+        )
+
+        if isinstance(e, subprocess.CalledProcessError):
+            error_msg += f"\nStderr: {e.stderr.strip()}"
+
+        error_msg += f"\n{error_header}\n"
+
+        logger.critical(error_msg)
+        sys.stderr.write(error_msg)
+        sys.stderr.flush()
+
+        # Re-raise the exception so the caller knows it failed.
+        # DVC might catch this and swallow it, but we've done our duty to notify.
+        raise
+
+    token = result.stdout.strip()
+    if not token:
+        raise ValueError("Command executed successfully but returned an empty token.")
+    return token
+
+
+class BearerAuthClient(httpx.Client):
```

Comment:
Do we need to extend `httpx.Client`? Could we make this work with custom [`httpx.Auth`](https://www.python-httpx.org/advanced/authentication/)?